crates/red_knot_project/tests/check.rs

```diff
@@ -198,10 +198,10 @@ impl SourceOrderVisitor<'_> for PullTypesVisitor<'_> {
     fn visit_stmt(&mut self, stmt: &Stmt) {
         match stmt {
             Stmt::FunctionDef(function) => {
-                let _ty = function.ty(&self.model);
+                let _ty = function.inferred_type(&self.model);
```

Comment:
We may want to wait for @carljm's feedback. I could see how he objects to `inferred_type` because it does return the declared type for classes or when the expression is in an annotation position...

Reply:
If so, then we should probably also change the documentation of the trait:
```rs
pub trait HasType {
    /// Returns the inferred type of `self`.
    ///
    /// ## Panics
    /// May panic if `self` is from another file than `model`.
    fn inferred_type<'db>(&self, model: &SemanticModel<'db>) -> Type<'db>;
}
```
I would think that maybe here in `semantic_model.rs`, we may not want to distinguish between declared/inferred type. They're all types that are generated by "Type inference".

But yes, I'll leave this open for further discussion.

Reply:
> I would think that maybe here in `semantic_model.rs`, we may not want to distinguish between declared/inferred type. They're all types that are generated by "Type inference".

Yes, I agree with this and I'm fine with the name changes in this PR.